src/maplayer.c

```diff
@@ -1067,7 +1069,8 @@ int msLayerWhichItems(layerObj *layer, int get_all, const char *metadata) {
   if (layer->filteritem)
     nt++;
   if (layer->styleitem && (strcasecmp(layer->styleitem, "AUTO") != 0) &&
-      (strncasecmp(layer->styleitem, "javascript://", 13) != 0))
+      STARTS_WITH_CI(layer->styleitem, "javascript://") &&
+      STARTS_WITH_CI(layer->styleitem, "sld://"))
```

Comment:
```suggestion
      !STARTS_WITH_CI(layer->styleitem, "javascript://") &&
      !STARTS_WITH_CI(layer->styleitem, "sld://"))
```

Reply:
Missed these comments, but eventually came to the same conclusion! Thanks for all the assistance @rouault. I think I messed up the alias logic, but will rebase on to #7035 and review tomorrow.